### PR TITLE
Add `convolve` iteration algorithm with tests

### PR DESCRIPTION
**Pull request by guillaume-chevalier**
Bonjour, 

I added the `convolve` iteration algorithm with tests: a simple way to do convolutions on lists. 

- [x] **If creating a new file :**
  - [x] added links to it in the README files ?
  - [x] included tests with it ?
  - [x] added description (overview of algorithm, time and space compleixty, and possible edge case) in docstrings ?

- [ ] ~~**if done some changes :**~~
  - [ ] ~~wrote short description in the PR explaining what the changes do ?~~
  - [ ] ~~Fixes #[issue number] if related to any issue~~

- [ ] ~~**other**~~

### CODE REVIEW

**Comment by guillaume-chevalier** [comment]
Note: the test `test_plus_one_v3` fails and was already failing before opening this PR, everything else works: 

```
py36 runtests: commands[0] | python3 -m unittest discover tests
.................E...............................................................................................................................................................................................................................
======================================================================
ERROR: test_plus_one_v3 (test_array.TestPlusOne)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/keon/algorithms/tests/test_array.py", line 252, in test_plus_one_v3
    self.assertListEqual(plus_one_v3([0]), [1])
  File "/home/travis/build/keon/algorithms/algorithms/arrays/plus_one.py", line 45, in plus_one_v3
    num_arr[idx] = (num_arr[idx] + 1) % 10
TypeError: list indices must be integers, not tuple
----------------------------------------------------------------------
Ran 241 tests in 0.433s
FAILED (errors=1)
ERROR: InvocationError for command '/home/travis/build/keon/algorithms/.tox/py36/bin/python3 -m unittest discover tests' (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   py36: commands failed
The command "tox -e $TOX_ENV" exited with 1.
0.45s$ pip3 uninstall -y algorithms
Uninstalling algorithms-0.1.0:
  Successfully uninstalled algorithms-0.1.0
```

File: algorithms/iterables/convolved.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            convolution will add some padding according to the kernel_size, and `VALID` is the same as
+            specifying `padding=0`.
+        default_value: Default fill value for padding and values outside iteration range.
+
+    For more information, refer to: 
+    - https://github.com/guillaume-chevalier/python-conv-lib/blob/master/conv/conv.py
+    - https://github.com/guillaume-chevalier/python-conv-lib
+    - MIT License, Copyright (c) 2018 Guillaume Chevalier
+    """
+    return conv.convolved(
```
Comment (keon) [inline algorithms/iterables/convolved.py]: Thanks for your contribution!
We refrain from using any third party libraries. If possible, please implement everything within this project.

File: tests/test_iterables.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
+from algorithms.iterables import (
+    convolved
+)
+
+import unittest
+
+
+class TestConvolved1D(unittest.TestCase):
```
Comment (keon) [inline tests/test_iterables.py]: the unit tests for `conv` should have been covered within the `conv` library. so this is invalid

File: algorithms/iterables/convolved.py (hunk 1 above)
Comment (guillaume-chevalier) [reply algorithms/iterables/convolved.py]: I won't edit this. Is this fine?

File: tests/test_iterables.py (hunk 2 above)
Comment (guillaume-chevalier) [reply tests/test_iterables.py]: Sure, those tests are the same as in the library, and copied. So we should just remove them from here?

**Comment by guillaume-chevalier** [comment]
Thanks for the review and for your time!

**Comment by guillaume-chevalier** [comment]
I made the changes.

**Comment by guillaume-chevalier** [comment]
@keon I finally decided to make the requested changes.

**Pull request merged**